File: P6.java
public class Rectangulo {
    private String color;
    private double base;
    private double altura;

    public Rectangulo(String color, double base, double altura) {
        this.color = color;
        this.base = base;
        this.altura = altura;
    }

    public String getColor() {
        return color;
    }

    public double getBase() {
        return base;
    }

    public double getAltura() {
        return altura;
    }

    public double getArea() {
        return area();
    }

    public double getPerimetro() {
        return perimetro();
    }

    private double perimetro() {
        return (base * 2) + (altura * 2);
    }

    private double area() {
        return base * altura;
    }

}


public class Cuadrado {
    private String color;
    private double lado;

    public Cuadrado(String color, double lado) {
        this.color = color;
        this.lado = lado;
    }

    public String getColor() {
        return color;
    }

    public double getLado() {
        return lado;
    }

    public double getArea() {
        return area();
    }

    public double getPerimetro() {
        return perimetro();
    }

    private double perimetro() {
        return 4 * lado;
    }

    private double area() {
        return lado * lado;
    }
}

public class Triangulo {
    private String color;
    private double base;
    private double altura;
    private double lado;

    public Triangulo(String color, double base, double altura, double lado) {
        this.color = color;
        this.base = base;
        this.altura = altura;
        this.lado = lado;
    }

    public String getColor() {
        return color;
    }

    public double getBase() {
        return base;
    }

    public double getAltura() {
        return altura;
    }

    public double getLado(){
        return lado;
    }

    public double getArea() {
        return area();
    }

    public double getPerimetro() {
        return perimetro();
    }

    private double area() {
        return (base * altura) /2;
    }

    private double perimetro() {
        return 3 * lado;
    }
}

public class Rombo {
    private String color;
    private double diagonalMayor;
    private double diagonalMenor;
    private double lado;

    public Rombo(String color, double diagonalMayor, double diagonalMenor, double lado) {
        this.color = color;
        this.diagonalMayor = diagonalMayor;
        this.diagonalMenor = diagonalMenor;
        this.lado = lado;
    }

    public String getColor() {
        return color;
    }

    public double getDiagonalMayor() {
        return diagonalMayor;
    }

    public double getDiagonalMenor() {
        return diagonalMenor;
    }

    public double getLado(){
         return lado;
    }

    public double getArea() {
        return area();
    }

    public double getPerimetro() {
        return perimetro();
    }

    private double perimetro() {
        return 4 * lado;
    }

    private double area() {
        return (diagonalMayor * diagonalMenor) / 2;
    }

}

public class Main{
     public static void main(String[] args) {
        Rectangulo rectangulo = new Rectangulo("azul", 5, 3);
        System.out.println("Color: " + rectangulo.getColor());
        System.out.println("Base: " + rectangulo.getBase());
        System.out.println("Altura: " + rectangulo.getAltura());
        System.out.println("Área: " + rectangulo.getArea());
        System.out.println("Perímetro: " + rectangulo.getPerimetro());

        Cuadrado cuadrado = new Cuadrado("amarillo", 8);
        System.out.println("Color: " + cuadrado.getColor());
        System.out.println("Lado: " + cuadrado.getLado());
        System.out.println("Área: " + cuadrado.getArea());
        System.out.println("Perímetro: " + cuadrado.getPerimetro());


        Triangulo triangulo = new Triangulo("azul", 5, 3,4);
        System.out.println("Color: " + triangulo.getColor());
        System.out.println("Base: " + triangulo.getBase());
        System.out.println("Altura: " + triangulo.getAltura());
        System.out.println("Lado; " + triangulo.getLado());
        System.out.println("Área: " + triangulo.getArea());
        System.out.println("Perímetro: " + triangulo.getPerimetro());

        Rombo rombo = new Rombo("blanco", 8, 4, 5);
        System.out.println("Color: " + rombo.getColor());
        System.out.println("Diagonal Mayor: " + rombo.getDiagonalMayor());
        System.out.println("Diagonal menor: " + rombo.getDiagonalMenor());
        System.out.println("Lado" + rombo.getLado());
        System.out.println("Área: " + rombo.getArea());
        System.out.println("Perímetro: " + rombo.getPerimetro());
 }
}

File: P6.py
import math

class Circulo:
    def __init__(self, color, radio):
        self.__color = color
        self.__radio = radio

    def getColor(self):
       return self.__color

    def getRadio(self):
       return self.__radio

    def getArea(self):
       return self.__area()

    def getPerimetro(self):
       return self.__perimetro()

    def __area(self):
       return math.pi * self.__radio**2

    def __perimetro(self):
       return 2 * math.pi * self.__radio

class Trapecio:
    def __init__(self, color, base_mayor, base_menor, altura, lado):
        self.__color = color
        self.__base_mayor = base_mayor
        self.__base_menor = base_menor
        self.__altura = altura
        self.__lado = lado

    def getColor(self):
        return self.__color

    def getBaseMayor(self):
        return self.__base_mayor

    def getBaseMenor(self):
        return self.__base_menor

    def getAltura(self):
        return self.__altura

    def getLado(self):
        return self.__lado

    def getArea(self):
        return self.__area()

    def getPerimetro(self):
        return self.__perimetro()

    def __area(self):
        return ((self.__base_mayor + self.__base_menor) / 2) * self.__altura

    def __perimetro(self):
        return self.__base_mayor + self.__base_menor + self.__lado * 2

class Hexagono:
    def __init__(self, color, apotema, lado):
        self.__color = color
        self.__apotema = apotema
        self.__lado = lado

    def getColor(self):
        return self.__color

    def getApotema(self):
        return self.__apotema

    def getLado(self):
        return self.__lado

    def getPerimetro(self):
        return self.__perimetro()

    def getArea(self):
        return self.__area()

    def __perimetro(self):
        return 6 * self.__lado

    def __area(self):
        return (self.__perimetro() * self.__apotema) /  2


#CIRCULO
circulo = Circulo("rojo", 5)
print("===CIRCULO=====")
print("Color", circulo.getColor())
print("Radio", circulo.getRadio())
print("Area", circulo.getArea())
print("Perimetro", circulo.getPerimetro())

#TRAPECIO
trapecio = Trapecio("verde", 10, 7, 6, 5)
print("====TRAPECIO====")
print("Color", trapecio.getColor())
print("Base mayor", trapecio.getBaseMayor())
print("Base menor", trapecio.getBaseMenor())
print("Altura", trapecio.getAltura())
print("Lateral", trapecio.getLado())
print("Area", trapecio.getArea())
print("Perimetro", trapecio.getPerimetro())

#HEXAGONO
hexagono = Hexagono("morado", 10, 5)
print("====HEXAGONO===")
print("Color", hexagono.getColor())
print("Apotema", hexagono.getApotema())
print("Lado", hexagono.getLado())
print("Area", hexagono.getArea())
print("Perimetro", hexagono.getPerimetro())
